src/SFML/Window/WindowImpl.cpp

```diff
@@ -175,42 +176,59 @@ void WindowImpl::setMaximumSize(const std::optional<Vector2u>& maximumSize)
 
 
 ////////////////////////////////////////////////////////////
-Event WindowImpl::popEvent(bool block)
+Event WindowImpl::popEventImpl()
 {
+    Event event;
+
+    if (!m_events.empty())
+    {
+        event = m_events.front();
+        m_events.pop();
+    }
+
+    return event;
+}
+
+
+////////////////////////////////////////////////////////////
+Event WindowImpl::popEventBlocking(Time timeout)
+{
+    const auto startTime = std::chrono::steady_clock::now();
+
+    const auto timedOut = [&]
```

Comment:
```suggestion
    const auto timedOut = [timeout, startTime = std::chrono::steady_clock::now()]
```
I'd prefer with be explicit with lambda captures.

Reply:
This lambda does not outlive the scope of the function it's defined in, so it's entirely safe to use `[&]` and it would be overly verbose to provide the captures explicitly. However, I like how using generalized lambda captures reduces the scope of `startTime`, so I will add that in.